pkg/frontend/querymiddleware/stats.go

```diff
@@ -55,26 +63,32 @@ func (s queryStatsMiddleware) Do(ctx context.Context, req Request) (Response, er
 		s.nonAlignedQueries.Inc()
 	}
 
-	if expr, err := parser.ParseExpr(req.GetQuery()); err == nil {
-		for _, selectors := range parser.ExtractSelectors(expr) {
-			for _, matcher := range selectors {
-				if matcher.Type != labels.MatchRegexp && matcher.Type != labels.MatchNotRegexp {
-					continue
-				}
-
-				s.regexpMatcherCount.Inc()
-				if matcher.IsRegexOptimized() {
-					s.regexpMatcherOptimizedCount.Inc()
-				}
-			}
-		}
-	}
-
+	s.regexpCounters(req)
+	s.consistencyCounters(ctx)
 	s.populateQueryDetails(ctx, req)
 
 	return s.next.Do(ctx, req)
 }
 
+func (s queryStatsMiddleware) regexpCounters(req Request) {
```

Comment:
[nit] Have you considered calling it `track....()` and `consistencyCounters` -> `track...()` too?

Reply:
do you mean something like `trackReadConsistency` and `trackRegexpMatchers`? 👍 done in d98612e34a75bd82127810c3601c2a8cb7e53a74